### PR TITLE
encryption: add kube-storage-version-migrator support

### PR DESCRIPTION
**Pull request by sttts**
This adds support for kube-storage-version-migrator, but does not switch to it. It just refactors and adds another migration for kube-storage-version-migrator. Each apiserver operator has to instantiate the migrator it wants.

### CODE REVIEW

**Comment by sttts** [comment]
/hold

until proven in kube-apiserver and openshift-apiserver.

File: pkg/operator/encryption/controllers/migrators/kubestorageversionmigrator.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/tools/cache"
+)
+
+const writeKeyAnnotationKey = "encryption.apiserver.operator.openshift.io/write-key"
+
+func NewKubeStorageVersionMigrator(client kubemigratorclient.Interface, informer migrationv1alpha1informer.Interface, discoveryClient discovery.ServerResourcesInterface) *KubeStorageVersionMigrator {
+	m := &KubeStorageVersionMigrator{
```
Comment (mfojtik) [inline pkg/operator/encryption/controllers/migrators/kubestorageversionmigrator.go]: nit: just return

File: pkg/operator/encryption/controllers/migrators/kubestorageversionmigrator.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	return m
+}
+
+// KubeStorageVersionMigrator runs migration through the kube-storage-version-migrator components,
+// driven by CustomResources.
+type KubeStorageVersionMigrator struct {
+	discoveryClient discovery.ServerResourcesInterface
+	client          kubemigratorclient.Interface
+	informer        migrationv1alpha1informer.Interface
+	ns              string
```
Comment (mfojtik) [inline pkg/operator/encryption/controllers/migrators/kubestorageversionmigrator.go]: is this targetNamespace ?

File: pkg/operator/encryption/controllers/migrators/kubestorageversionmigrator.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+				Resource: gr.Resource,
+			},
+		},
+	})
+
+	return false, nil, err
+}
+
+func (m *KubeStorageVersionMigrator) PruneMigration(gr schema.GroupResource) error {
+	name := migrationResourceName(gr)
```
Comment (mfojtik) [inline pkg/operator/encryption/controllers/migrators/kubestorageversionmigrator.go]: nit: do we need this as var?

File: pkg/operator/encryption/controllers/migrators/kubestorageversionmigrator.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+	if err := m.client.MigrationV1alpha1().StorageVersionMigrations().Delete(name, &metav1.DeleteOptions{}); err != nil && !errors.IsNotFound(err) {
```
Comment (mfojtik) [inline pkg/operator/encryption/controllers/migrators/kubestorageversionmigrator.go]: nit: just return

File: pkg/operator/encryption/controllers/migrators/kubestorageversionmigrator.go (hunk 1 above)
Comment (sttts) [reply pkg/operator/encryption/controllers/migrators/kubestorageversionmigrator.go]: Fixed

File: pkg/operator/encryption/controllers/migrators/kubestorageversionmigrator.go (hunk 2 above)
Comment (sttts) [reply pkg/operator/encryption/controllers/migrators/kubestorageversionmigrator.go]: unused. removed.

File: pkg/operator/encryption/controllers/migrators/kubestorageversionmigrator.go (hunk 3 above)
Comment (sttts) [reply pkg/operator/encryption/controllers/migrators/kubestorageversionmigrator.go]: it's used 3 times

File: pkg/operator/encryption/controllers/migrators/kubestorageversionmigrator.go (hunk 4 above)
Comment (sttts) [reply pkg/operator/encryption/controllers/migrators/kubestorageversionmigrator.go]: no. NotFound is special.

**Comment by sttts** [comment]
Skipping e2e-aws-encryption as it was green twice before without touching and non-unit-test code.

**Pull request merged**